modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java

```diff
@@ -376,9 +389,10 @@ static Result analyze(Query query, Version indexVersion) {
                                 msm += result.minimumShouldMatch;
                             }
                             verified &= result.verified;
+                            matchAllDocs &= result.matchAllDocs;
                             extractions.addAll(result.extractions);
                         }
-                        return new Result(verified, extractions, msm);
+                        return new Result(matchAllDocs, verified, extractions, msm);
```

Comment:
the fact that we do not look at the value of msm for matchAllDocs looks like a bug

Reply:
So the reason that this works at all, is that because the `` clause it not part of the `CoveringQuery` itself, but the parent bool query wrapping the `CoveringQuery`, see PercolatorFieldMapper.java line 299. But I agree it looks off, so I'll take msm also in account for matchAllDocs queries.

Reply:
wait, I'll do what you mentioned in the second comment.